src/main/java/blockrenderer6343/integration/gregtech/GT_GUI_MultiblocksHandler.java

```diff
@@ -136,6 +150,22 @@ private void projectMultiblocks() {
         baseWorld.removeTileEntity(lookingPos.blockX, lookingPos.blockY + 1, lookingPos.blockZ);
     }
 
+    private void neiOverlay() {
+        var recipeInputs = new NBTTagCompound();
+        var currentScreen = (GuiRecipe) Minecraft.getMinecraft().currentScreen;
+        Minecraft.getMinecraft().displayGuiScreen(currentScreen.firstGui);
+        var inputIndex = 0;
+        for (ItemStack itemStack : ingredients) {
+            if (itemStack != null) {
+                NBTTagCompound itemStackNBT = new NBTTagCompound();
+                itemStack.writeToNBT(itemStackNBT);
+                itemStackNBT.setInteger("Count", itemStack.stackSize);
+                recipeInputs.setTag("#" + inputIndex++, itemStackNBT);
+            }
+        }
```

Comment:
Overall don't use var if possible.

Instead of using an enhanced for loop, you could use the c-style for loop so that you don't need to keep the inputIndex variable separate from the loop.